spark/src/test/scala/org/apache/spark/sql/delta/skipping/clustering/ClusteredTableDDLSuite.scala

```diff
@@ -666,6 +673,60 @@ trait ClusteredTableDDLSuiteBase
       }
     }
   }
+
+  testQuietly("validate CLONE on clustered table") {
+    import testImplicits._
+    val srcTable = "SrcTbl"
+    val dstTable = "DestTbl"
+    Seq(true, false).foreach { useSQL =>
+      withTable(srcTable, dstTable) {
+        // Create the source table.
+        sql(s"CREATE TABLE $srcTable (col1 INT, col2 INT, col3 INT) " +
+          s"USING delta CLUSTER BY (col1, col2)")
+        val tableIdent = new TableIdentifier(srcTable)
+        (1 to 100).map(i => (i, i + 1000, i + 100)).toDF("col1", "col2", "col3")
+          .repartitionByRange(100, col("col1"))
+          .write.format("delta").mode("append").saveAsTable(srcTable)
+
+        // Force clustering on the source table.
+        val (_, srcSnapshot) = DeltaLog.forTableWithSnapshot(spark, tableIdent)
+        val ingestionSize = srcSnapshot.allFiles.collect().map(_.size).sum
+        withSQLConf(
+          DeltaSQLConf.DELTA_OPTIMIZE_MAX_FILE_SIZE.key -> (ingestionSize / 4).toString) {
+          runOptimize(srcTable) { res =>
+            assert(res.numFilesAdded === 4)
+            assert(res.numFilesRemoved === 100)
+          }
+        }
+
+        // Create destination table as a clone of the source table.
+        if (useSQL) {
+          sql(s"CREATE TABLE $dstTable SHALLOW CLONE $srcTable")
+        } else {
+          val sourceIdentifier = Identifier.of(
+            TableIdentifier(srcTable).database.toArray, TableIdentifier(srcTable).table)
+          val sourceRelation = DataSourceV2Relation.create(
+            DeltaTableV2(spark, TableIdentifier(srcTable), ""), None, Some(sourceIdentifier))
+          val clone = CloneTableStatement(
```

Comment:
We should test the external API `DeltaTable.clone` instead of the internal implementation, ref: https://docs.delta.io/latest/api/scala/spark/io/delta/tables/DeltaTable.html#clone():Object